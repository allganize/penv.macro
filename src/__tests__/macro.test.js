import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'

process.env.NODE_ENV = 'test'

pluginTester({
  plugin,
  snapshot: true,
  tests: {
    'should replace the environment variable to the matched value': {
      code: `
        import env from './src/macro';

        const variable = env({
          development: 'development',
          staging: 'staging',
          production: 'production',
          test: () => {
            console.log('test');
            
            return 'test';
          }
        });
      `,
    },

    'should use `null` as default value if no relevant value was matched': {
      code: `
        import inlineEnv from './src/macro';

        const variable = inlineEnv({
          development: 'development',
          staging: 'staging',
          production: 'production',
        });
      `,
    },

    'should remove expression if `env` is a standalone expression': {
      // only: true,
      code: `
        import penv from './src/macro';

        penv({
          unmatched: () => {},
        });
      `,
    },
  },
})
