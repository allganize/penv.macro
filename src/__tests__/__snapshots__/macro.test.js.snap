// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros should remove expression if \`env\` is a standalone expression: should remove expression if \`env\` is a standalone expression 1`] = `
"
import penv from './src/macro';

penv({
  unmatched: () => {},
});

      ↓ ↓ ↓ ↓ ↓ ↓


"
`;

exports[`macros should replace the environment variable to the matched value: should replace the environment variable to the matched value 1`] = `
"
import env from './src/macro';

const variable = env({
  development: 'development',
  staging: 'staging',
  production: 'production',
  test: () => {
    console.log('test');
    
    return 'test';
  }
});

      ↓ ↓ ↓ ↓ ↓ ↓

const variable = () => {
  console.log('test');

  return 'test';
};
"
`;

exports[`macros should use \`null\` as default value if no relevant value was matched: should use \`null\` as default value if no relevant value was matched 1`] = `
"
import inlineEnv from './src/macro';

const variable = inlineEnv({
  development: 'development',
  staging: 'staging',
  production: 'production',
});

      ↓ ↓ ↓ ↓ ↓ ↓

const variable = null;
"
`;
